ironfist/src/cpp/combat/combat.cpp

```diff
@@ -613,80 +612,67 @@ void army::LoadResources()
 		this->combatSounds[4] = gpResourceManager->GetSample(gText);
-		if (this->creature.creature_flags & SHOOTER)
-		{
+		if (this->creature.creature_flags & SHOOTER) {
 			sprintf(gText, "%sshot.82M", this->creature.short_name);
 			this->combatSounds[3] = gpResourceManager->GetSample(gText);
 		}
-		if (this->creatureIdx == CREATURE_VAMPIRE || this->creatureIdx == CREATURE_VAMPIRE_LORD)
-		{
-			sprintf(gText, "%sext1.82M", this->creature.short_name);
-			this->combatSounds[5] = gpResourceManager->GetSample(gText);
-			sprintf(gText, "%sext2.82M", this->creature.short_name);
-			this->combatSounds[6] = gpResourceManager->GetSample(gText);
-		}
-		if (this->creatureIdx == CREATURE_LICH || this->creatureIdx == CREATURE_POWER_LICH)
-		{
-			sprintf(gText, "%sexpl.82M", this->creature.short_name);
-			this->combatSounds[5] = gpResourceManager->GetSample(gText);
+		switch (creatureID) {
+			case CREATURE_VAMPIRE: case CREATURE_VAMPIRE_LORD: {
+				sprintf(gText, "%sext1.82M", this->creature.short_name);
+				this->combatSounds[5] = gpResourceManager->GetSample(gText);
+				sprintf(gText, "%sext2.82M", this->creature.short_name);
+				this->combatSounds[6] = gpResourceManager->GetSample(gText);
+				break;
+			}
+			case CREATURE_LICH: case CREATURE_POWER_LICH: {
+				sprintf(gText, "%sexpl.82M", this->creature.short_name);
+				this->combatSounds[5] = gpResourceManager->GetSample(gText);
+			}
 		}
-		this->creatureIcon = gpResourceManager->GetIcon(cMonFilename[this->creatureIdx]);
-		if (this->creature.creature_flags & SHOOTER)
-		{
-			if (this->creatureIdx != CREATURE_GIANT && this->creatureIdx != CREATURE_TITAN)
-			{
-				if (this->creatureIdx == CREATURE_HALFLING)
-				{
+
+		this->creatureIcon = gpResourceManager->GetIcon(cMonFilename[creatureID]);
+		// Loading projectiles
+		if (this->creature.creature_flags & SHOOTER) {
+			switch (creatureID) {
+				case CREATURE_HALFLING: {
 					sprintf(gText, "halflmsl.icn");
+					break;
 				}
-				else if (this->creatureIdx != CREATURE_ARCHER && this->creatureIdx != CREATURE_RANGER)
-				{
-					if (this->creatureIdx != CREATURE_LICH && this->creatureIdx != CREATURE_POWER_LICH)
-					{
-						if (this->creatureIdx != CREATURE_ORC && this->creatureIdx != CREATURE_ORC_CHIEF)
-						{
-							if (this->creatureIdx != CREATURE_DRUID && this->creatureIdx != CREATURE_GREATER_DRUID)
-							{
-								if (this->creatureIdx != CREATURE_TROLL && this->creatureIdx != CREATURE_WAR_TROLL)
-									sprintf(gText, "elf__msl.icn");
-								else
-									sprintf(gText, "trollmsl.icn");
-							}
-							else
-							{
-								sprintf(gText, "druidmsl.icn");
-							}
-						}
-						else
-						{
-							sprintf(gText, "orc__msl.icn");
-						}
-					}
-					else
-					{
-						sprintf(gText, "lich_msl.icn");
-					}
+				case CREATURE_GIANT: case CREATURE_TITAN: {
+					sprintf(gText, "titanmsl.icn");
+					break;
 				}
-				else
-				{
+				case CREATURE_ARCHER: case CREATURE_RANGER: {
```

Comment:
I've never seen the pattern of surrounding a case statement in braces, but I like it. Thanks for showing this to me.